cli/lib/nf3.mjs

```diff
@@ -1439,6 +1441,36 @@ class Nf3 {
     });
     return res.data.isWhitelisted;
   }
+
+  /**
+   Validates an X509 (RSA) certificate
+   */
+  async validateCertificate(certificate, ethereumAddress, derPrivateKey) {
```

Comment:
No problem!:
- Yes, type is Buffer but I suspect a string will also work.  The web3js api will probably work it out.
- There's no problem, you can whitelist as many addresses as you like. It's not free to do that though, so you would probably want to limit how many you register.
- So, your certificate contains a public key, which the certificate attests to.  derPrivateKey is the corresponding private key to the public key in the certificate.  It's nothing to do with your Ethereum private key. It needs to be available to Nf3 so that Nf3 can use it to sign your Ethereum address.  This action proves that the owner of the Ethereum address is also the person attested to in the certificate.